### ISSUE TITLE
Accessibility: support aria-pressed attributes

### ISSUE THREAD

**Issue by anshuman-ms** [xyflow/xyflow#4264]
### Please describe the feature that you want to propose

Based on Accessibility guidelines, buttons should not have interactive controls in it. https://dequeuniversity.com/rules/axe/4.5/nested-interactive

So, if we add any buttons or interactions using a custom node, then the accessibility compliance is hit. I checked with couple of my colleagues, and having a pressed state, might help. 

When a node is selected it can probably be marked as aria-pressed="true" then the controls inside become interactive or else one other solution could be is to allow custom nodes to fully take over the a11y requirements. That means, not marking the node as button and letting the control inside handle accessibility. This is not possible today as selection is done via the role="button" div which is rendered and managed by React flow.

Any guidance or support will be appreciated

**Comment by anshuman-ms** [comment]
@moklick I tried that, the problem here is the if you turn off  focus with `nodedFocusable={false}` then the ability to tab goes away and creates a much bigger a11y issue. Is there a way to pass these attributes or not render a button but a div?

**Comment by vanch3d** [comment]
This has been a long hair-pulling issue for me as well :-)
The short answer is that you cannot add ANY control within a node. All the "interactive" examples in the `React Flow` pages (e.g. the one on the main landing page) violate that "nested interactive control" rule, see https://dequeuniversity.com/rules/axe/4.2/nested-interactive

The long answer however is that something could be tried. 
There is an example where the "drag" handle is restricted to a sub-part of the node, see https://reactflow.dev/examples/nodes/drag-handle
In the example, the node retains its `[role="button"] attribute, making it receive the focus for the whole node, preventing nesting

If the `[role="button"]` was added to the custom handle and removed from the whole node, then you might be able to add controls: 
- they will be within the DOM tree of the node
- they won't be nested within the Accessibility tree of the node
But it means refactoring the internal code (the node wrapper is not accessible for customisation, only it's content)

**Comment by moklick** [comment]
Now that I look at this issue again, a better default for nodes and edges might be:

role="group"
aria-roledescription="node" / "edge"

We need to check if this could solve this.

**Comment by moklick** [comment]
Since React Flow 12.7.0 nodes and edges use "group" as a role. You can change this by using the new `ariaRole` option. Thanks @printerscanner 🔥
